src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -0,0 +1,215 @@
+        } else if (StringUtils.equalsIgnoreCase(line.getEmployeeStatus(), ConcurConstants.NON_EMPLOYEE_STATUS_CODE)) {
+            payeeIdEntry.setIdType(ConcurConstants.NON_EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else {
+            LOG.error("buildPayeeIdEntry, Unable to to set the payee ID type based do the line's employee status " + line.getEmployeeStatus());
+        }
+        return payeeIdEntry;
+    }
+
+    @Override
+    public PdpFeedDetailEntry buildPdpFeedDetailEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedDetailEntry currentDetailEntry  = new PdpFeedDetailEntry();
+        currentDetailEntry.setSourceDocNbr(buildSourceDocumentNumber(line.getReportId()));
+        currentDetailEntry.setFsOriginCd(getConcurParameterValue(ConcurParameterConstants.CONCUR_AP_PDP_ORIGINATION_CODE));
+        currentDetailEntry.setFdocTypCd(getConcurParameterValue(ConcurParameterConstants.CONCUR_SAE_PDP_DOCUMENT_TYPE));
+        currentDetailEntry.setInvoiceNbr(StringUtils.EMPTY);
+        currentDetailEntry.setPoNbr(StringUtils.EMPTY);
+        currentDetailEntry.setInvoiceDate(formatDate(line.getBatchDate()));
+        List<String> paymentTextStrings = new ArrayList<String>();
+        paymentTextStrings.add("Report ID: " + line.getReportId());
+        currentDetailEntry.setPaymentText(paymentTextStrings);
+        return currentDetailEntry;
+    }
+    
+    @Override
+    public String buildSourceDocumentNumber(String reportId) {
+        String sourceDocNumber = getConcurParameterValue(ConcurParameterConstants.CONCUR_SAE_PDP_DOCUMENT_TYPE) + reportId;
+        return StringUtils.substring(sourceDocNumber, 0, ConcurConstants.SOURCE_DOCUMENT_NUMBER_FIELD_SIZE);
+    }
+
+    @Override
+    public PdpFeedAccountingEntry buildPdpFeedAccountingEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedAccountingEntry currentAccountingEntry =  new PdpFeedAccountingEntry();
+        currentAccountingEntry.setCoaCd(line.getChartOfAccountsCode());
+        currentAccountingEntry.setAccountNbr(line.getAccountNumber());
+        currentAccountingEntry.setSubAccountNbr(line.getSubAccountNumber());
+        currentAccountingEntry.setObjectCd(line.getJournalAccountCode());
+        currentAccountingEntry.setSubObjectCd(line.getSubObjectCode());
+        currentAccountingEntry.setOrgRefId(line.getOrgRefId());
+        currentAccountingEntry.setProjectCd(line.getProjectCode());
+        currentAccountingEntry.setAmount(KualiDecimal.ZERO);
+        return currentAccountingEntry;
+    }
+    
+    @Override
+    public PdpFeedTrailerEntry buildPdpFeedTrailerEntry(PdpFeedFileBaseEntry pdpFeedFileBaseEntry, KualiDecimal pdpTotal) {
+        PdpFeedTrailerEntry trailerEntry = new PdpFeedTrailerEntry();
+        int numberOfDetails = 0;
+        for (PdpFeedGroupEntry group : pdpFeedFileBaseEntry.getGroup()) {
+            numberOfDetails += group.getDetail().size();
```

Comment:
I don't think it is needed, it will just add another element to keep track of, and another input value to this function.  Per clean code, functions are best with a small number of input values.  As an aside, we validate the sums in other areas.